arcgis-ios-sdk-samples/Layers/Identify raster cell/IdentifyRasterCellViewController.swift

```diff
@@ -0,0 +1,133 @@
+    
+    /// The custom stack view to show in a callout.
+    @IBOutlet weak var calloutStackView: IdentifyRasterCellStackView!
+    
+    /// The map view managed by the view controller.
+    @IBOutlet weak var mapView: AGSMapView! {
+        didSet {
+            mapView.map = makeMap()
+            mapView.touchDelegate = self
+            mapView.callout.customView = calloutStackView
+            mapView.callout.isAccessoryButtonHidden = true
+        }
+    }
+    
+    // MARK: Properties
+    
+    /// The raster layer created using local raster file.
+    let rasterLayer = AGSRasterLayer(raster: AGSRaster(name: "SA_EVI_8Day_03May20", extension: "tif"))
+    /// A formatter for coordinates.
+    let formatter: NumberFormatter = {
+        let formatter = NumberFormatter()
+        formatter.numberStyle = .decimal
+        formatter.usesGroupingSeparator = false
+        formatter.minimumFractionDigits = 3
+        formatter.maximumFractionDigits = 3
+        return formatter
+    }()
+    
+    // MARK: Initialize map and utility network
+    
+    /// Create a map.
+    ///
+    /// - Returns: An `AGSMap` object.
+    func makeMap() -> AGSMap {
+        let map = AGSMap(basemapType: .oceans, latitude: -34.1, longitude: 18.6, levelOfDetail: 9)
+        map.operationalLayers.add(rasterLayer)
+        return map
+    }
+    
+    /// Identify a pixel on the raster layer and display the result in a callout.
+    ///
+    /// - Parameters:
+    ///   - screenPoint: The point at which to identify features.
+    ///   - offset: An offset to the point to identify to display the callout.
+    func identifyPixel(at screenPoint: CGPoint, offset: CGPoint = .zero) {
+        mapView.identifyLayer(rasterLayer, screenPoint: screenPoint, tolerance: 1.0, returnPopupsOnly: false) { [weak self] identifyResult in
+            guard let self = self else { return }
+            if identifyResult.geoElements.isEmpty {
+                // If there are no geoelements identified, e.g. not on a raster layer, dismiss the callout.
+                self.mapView.callout.dismiss()
+                return
+            }
+            guard let cell = identifyResult.geoElements.first(where: { $0 is AGSRasterCell }) else {
+                return
+            }
+            var calloutText: [String] = []
+            var xyCoordinates: [String] = []
+            cell.attributes.forEach { attribute in
+                calloutText.append("\(attribute.key): \(attribute.value)")
+            }
+            let x = ("X", cell.geometry!.extent.xMin)
+            let y = ("Y", cell.geometry!.extent.yMin)
+            [x, y].forEach { (name, value) in
+                xyCoordinates.append("\(name): \(self.formatter.string(from: value as NSNumber)!)")
+            }
+            
+            self.calloutStackView.attributesLabel.text = calloutText.joined(separator: "\n")
+            self.calloutStackView.coordinatesLabel.text = xyCoordinates.joined(separator: "\n")
```

Comment:
I fear I may have given you not-so-great advice last time. Now that I see better what is going on, take a look at this way and let me know what you think:

```suggestion
            let calloutText = cell.attributes
                .map { "\($0.key): \($0.value)" }
                .joined(separator: "\n")
            let extent = cell.geometry!.extent
            let xyCoordinates = """
            X: \(self.formatter.string(from: extent.xMin as NSNumber)!)
            Y: \(self.formatter.string(from: extent.yMin as NSNumber)!)
            """
            
            self.calloutStackView.attributesLabel.text = calloutText
            self.calloutStackView.coordinatesLabel.text = xyCoordinates
```

Note: The screenshot has blank space between `calloutText` and `xyCoordinates`. Instead of adding a newline as was there before, could you just adjust the spacing between the two labels?

Reply:
Sure! I'll fix that.